src/python/pants/engine/rules.py

```diff
@@ -470,7 +470,7 @@ def __init__(
     self._output_type = output_type
     self.input_selectors = input_selectors
     self.input_gets = input_gets
-    self.func = func
+    self.func = func  # type: ignore
```

Comment:
NB: these ignores will have error codes once we can land MyPy 0.730 / 0.740. This is currently blocked by the change to use implicit namespace packages in Pants.